Doppler.AccountPlans/Utils/CalculateUpgradeCostHelper.cs

```diff
@@ -7,19 +7,28 @@ public static class CalculateUpgradeCostHelper
     {
         public static PlanAmountDetails CalculatePlanAmountDetails(PlanInformation newPlan, PlanDiscountInformation newDiscount, PlanInformation currentPlan, DateTime now)
         {
-            var currentBaseMonth = now.Day >= 21 ? currentPlan.CurrentMonthPlan : currentPlan.CurrentMonthPlan - 1;
-            var totalPlan = newDiscount.MonthPlan != currentBaseMonth ? (newPlan.Fee - currentPlan.Fee) * (1 - (newDiscount.DiscountPlanFee / 100)) : newPlan.Fee * (1 - (newDiscount.DiscountPlanFee / 100));
 
-            return new PlanAmountDetails
+            var isMonthPlan = currentPlan.CurrentMonthPlan == 0;
+
+            var currentBaseMonth = now.Day < 21 ?
+                (currentPlan.CurrentMonthPlan - 1) :
+                (isMonthPlan ? 1 : currentPlan.CurrentMonthPlan);
```

Comment:
In my opining, writting this in this way is cleaner and shorter:

```suggestion
            var isMonthPlan = currentPlan.CurrentMonthPlan == 0;

            var currentBaseMonth = now.Day < 21 ? currentPlan.CurrentMonthPlan - 1 
                : isMonthPlan ? 1 
                : currentPlan.CurrentMonthPlan);
```

Or even:

```suggestion
            var currentBaseMonth = now.Day < 21 ? currentPlan.CurrentMonthPlan - 1 
                : currentPlan.CurrentMonthPlan == 0 ? 1 
                : currentPlan.CurrentMonthPlan);
```